src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -152,7 +154,12 @@ public DefaultLogger(OutputStream infoStream,
 
     @Override
     protected void finishLocalSetup() {
-        // No code by default
+        if (infoStreamOptions == null) {
```

Comment:
@Romani, as you mentioned above https://github.com/checkstyle/checkstyle/pull/14475/files#r1546355855, logic is null-safe. Hence, we can remove the validation. I assume you were talking about null-check. I moved them to finishLocalSetup. Or, null-check should be scrapped entirely?

Reply:
If null is not possible, we do not need null check.

Reply:
> If null is not possible, we do not need null check.

I am bit confused. I am initialising them but, i think they can be null. If anyone passes null to the constructor, i guess the values would be still null. Shouldn't we require an exception for that?

Reply:
If npe is possible, it is better to throw exact exception

Reply:
please remove such validation.
Logic is null safe.

Reply:
How do i design the test case which takes null value?

Reply:
you have junit test (not Input based) for null already, keep it, it will work as test for NONE.

Reply:
i have tried to modify the test case. please let me know your feedback.